vscode/microsoft-kiota/src/steps.ts

```diff
@@ -118,21 +119,21 @@ export async function searchSteps(searchCallBack: (searchQuery: string) => Thena
     return state;
 }
 
-export async function generateSteps(existingConfiguration: Partial<GenerateState>, languagesInformation?: LanguagesInformation, isDeeplinkEnabled?: boolean) {
+export async function generateSteps(existingConfiguration: Partial<GenerateState>, languagesInformation?: LanguagesInformation, deepLinkParams?: Partial<IntegrationParams>) {
     const state = { ...existingConfiguration } as Partial<GenerateState>;
     if (existingConfiguration.generationType && existingConfiguration.clientClassName && existingConfiguration.clientNamespaceName && existingConfiguration.outputPath && existingConfiguration.language &&
         typeof existingConfiguration.generationType === 'string' && existingConfiguration.clientNamespaceName === 'string' && typeof existingConfiguration.outputPath === 'string' && typeof existingConfiguration.language === 'string' &&
         existingConfiguration.generationType.length > 0 && existingConfiguration.clientClassName.length > 0 && existingConfiguration.clientNamespaceName.length > 0 && existingConfiguration.outputPath.length > 0 && existingConfiguration.language.length > 0) {
         return state;
     }
 
-    const isDeepLinkPluginNameProvided = isDeeplinkEnabled && state.pluginName;
-    const isDeepLinkGenerationTypeProvided = isDeeplinkEnabled && state.generationType;
-    const isDeepLinkPluginTypeProvided = isDeeplinkEnabled && state.pluginTypes;
-    const isDeepLinkLanguageProvided = isDeeplinkEnabled && state.language;
-    const isDeepLinkOutputPathProvided = isDeeplinkEnabled && state.outputPath;
-    const isDeepLinkClientClassNameProvided = isDeeplinkEnabled && state.clientClassName;
-
+    const deeplinkEnabled = !!deepLinkParams && isDeeplinkEnabled(deepLinkParams);
```

Comment:
Why are you using `!!` does the boolean coercion not work here?

Reply:
deepLinkParams could be undefined. So this checks first that it is truthy before checking that deeplinking is enabled

Reply:
```suggestion
    const deeplinkEnabled = deepLinkParams && isDeeplinkEnabled(deepLinkParams);
```

I meant this